internal/tuf/v01/root.go

```diff
@@ -444,27 +445,44 @@ func (r *RootMetadata) AddControllerRepository(name, location string, initialRoo
 		r.MultiRepository = &MultiRepository{ControllerRepositories: []*OtherRepository{}}
 	}
 
-	// TODO: check for duplicates
-
-	otherRepository := &OtherRepository{
-		Name:                  name,
-		Location:              location,
-		InitialRootPrincipals: make([]*Key, 0, len(initialRootPrincipals)),
+	for _, repo := range r.MultiRepository.ControllerRepositories {
+		if repo.Name == name || repo.Location == location {
+			return tuf.ErrDuplicateControllerRepository
+		}
 	}
 
+	newKeyIDs := make([]tuf.Principal, 0, len(initialRootPrincipals))
 	for _, principal := range initialRootPrincipals {
 		key, isKey := principal.(*Key)
 		if !isKey {
 			return tuf.ErrInvalidPrincipalType
 		}
+		newKeyIDs = append(newKeyIDs, key)
+	}
+
+	for _, repo := range r.MultiRepository.ControllerRepositories {
+		existingKeyIDs := make(map[string]struct{}, len(repo.InitialRootPrincipals))
```

Comment:
can we just use a `set` here? We have one included and I think it even may handle set equal comparison...

Reply:
https://github.com/gittuf/gittuf/blob/main/internal/common/set/set.go#L14-L16